orttraining/orttraining/training_ops/cpu/torch/torch_custom_function_kernel_base.cc

```diff
@@ -224,11 +224,14 @@ void PythonOpBase::SetOtherOutputs(OpKernelContext* context, std::vector<OrtValu
 void PythonOpGradBase::Init(const OpKernelInfo& info) {
   ORT_THROW_IF_ERROR(info.GetAttr("name", &name_));
   ORT_THROW_IF_ERROR(info.GetAttr("inplace", &inplace_));
-  ORT_THROW_IF_ERROR(info.GetAttrs("input_tensor_types", input_tensor_types_));
+  ORT_THROW_IF_ERROR(info.GetAttr("output_convention", &output_convention_));
   ORT_THROW_IF_ERROR(info.GetAttrs("output_tensor_types", output_tensor_types_));
-  input_tensor_requires_grads_ = info.GetAttrsOrDefault("input_tensor_requires_grads", std::vector<int64_t>());
   output_tensor_requires_grads_ = info.GetAttrsOrDefault("output_tensor_requires_grads", std::vector<int64_t>());
-  SetPositions();
+  ORT_ENFORCE(output_tensor_types_.size() == output_tensor_requires_grads_.size(), "backward tensor output count mismatch");
+
+  // Exclude context and forward activations from inputs
+  const size_t tensor_input_count = (info.node().InputDefs().size() - 1) / 2;
```

Comment:
This looks a bit fragile if we add extra inputs at some point in the future.  We could check that InputDefs().size() is odd as one safeguard -- are there other details we can assert?

Reply:
+1. Please mark a todo here. 
Also, we are including all the outputs as a work around for the seg fault issue. 
After fix, the number of input should be just N+1 .

Reply:
Thanks for the feedbacks, I would admit this is really not a good one.  Combining with Wei-Sheng's comment, I think a better way is, we add some redundant attributes like "input_tensor_types_" back to easily infer the tensor input count. That would be safe once we added more inputs. 

Besides, there are other pieces similar with this. I will try to eliminate this kind of logic once getting a better way to handle the lifecycle of ctx along with forward torch tensors.

Reply:
Added a todo on my head